Project21/Project21/Assignment 3.c

```diff
@@ -0,0 +1,73 @@
+// ------------------------------------------------------------------------------------------------
+// Assignment 3 ~ A training program for new joiners at Metamation, Batch - July 2024.
+// Copyright (c) Metamation India.
+// ------------------------------------------------------------------
+// Assignment 3
+// Decimal to Binary conversion and Decimal to HexaDecimal conversion
+// ------------------------------------------------------------------------------------------------
+#include<stdio.h>
+#include <malloc.h>
+#define INT_SIZE sizeof(int)*8
+#define HEX_SIZE 8
+int binCount = 0;
+
+/// <summary>Function to convert a decimal number to binary</summary>
+int DecimalToBinary (int n) { 
+   int temp = 0;
+   temp = n;
+   while (temp != 0) {
+      temp = temp / 2;
+      binCount++;
+   }
+   if (binCount % HEX_SIZE != 0) binCount = binCount + (HEX_SIZE - (binCount % HEX_SIZE));
+   int index = 0;
+   int* binary = (int*)malloc (binCount * sizeof (int)); 
+   int num = n;
+   index = binCount - 1; 
+   while (index >= 0) {
+      binary[index] = n & 1;// Extract the least significant bit (LSB) from 'n' 
+      index--;
+      n >>= 1;// Right-shift 'n' by one bit to remove the extracted bit 
+   }
+   printf ("Binary value: ");
+   for (int i = 0; i < binCount; i++)
+      printf ("%d", binary[i]);
+   printf ("\n");
+   binCount = 0;
+   return num;
+}
+
+/// <summary>Function to convert a decimal number to Hexadecimal</summary>
+void DecimalToHexadecimal (int num) {
+   char hex[HEX_SIZE + 1];// Array to store the hexadecimal digits 
+   unsigned int unsNum = (unsigned int)num; //Convert the input number to an unsigned integer 
+   char hexChars[] = "0123456789ABCDEF";// Extract the last 4 bits (1 hex digit) 
+   int index = HEX_SIZE;
+   for (int i = index - 1; i >= 0; i--) {
+      hex[i] = hexChars[unsNum & 0xF];
+      unsNum >>= 4;
+   }
+   hex[index] = '\0';// Null - terminate the hex string
+   printf ("Hexadecimal value: %s\n\n", hex);
+}
+
+int main () {
+   int dec;
+   char term;
+   while (1) {

```

Comment:
It's a nice way of prompting the user till a valid integer is received. But when will this while loop break and return 0?